### PR TITLE
Transform points in laser frame to robot frame

### LINKED ISSUE #274: System tests are flaky on main
### Bug description
The system tests are flaky; `git bisect` points directly to the laser scan rework.

**Platform (please complete the following information):**
 - OS: Ubuntu Jammy (ROS Humble)
 - `Beluga` version: 51576fbad2a16d6f42edc6dde43726390934bf63

### How to reproduce
1. `sudo apt update && sudo apt install parallel -y`
2. `colcon build --packages-up-to beluga_system_tests`
3. `parallel -N0 --joblog /tmp/log colcon test --packages-select beluga_system_tests --return-code-on-test-failure ::: {1..14}`
4. `cat /tmp/log`

**Expected behavior**
All runs should be successful (return code 0).

**Actual behavior**
About 10/14 successful runs.

### PR DESCRIPTION
**Pull request by nahueespinosa**
### Proposed changes

Closes #274.

#### Type of change

- [x] 🐛 Bugfix (change which fixes an issue)
- [ ] 🚀 Feature (change which adds functionality)
- [ ] 📚 Documentation (change which fixes or extends documentation)

### Checklist

- [x] Lint and unit tests (if any) pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] All commits have been signed for [DCO](https://developercertificate.org/)

### CODE REVIEW

File: beluga_amcl/src/amcl_node.cpp [hunk 1]
```diff
@@ -931,7 +931,10 @@ void AmclNode::laser_callback(ExecutionPolicy&& exec_policy, sensor_msgs::msg::L
           get_parameter("laser_max_range").as_double(),
       }
               .points_in_cartesian_coordinates() |
-          ranges::views::transform([](const auto& value) { return std::make_pair(value.x(), value.y()); }) |
+          ranges::views::transform([&](const auto& p) {
+            const auto result = base_to_laser_transform * Sophus::Vector3d{p.x(), p.y(), 0};
```
Comment (glpuga) [inline beluga_amcl/src/amcl_node.cpp:935]: nit: good opportunity to rename to `laser_pose_in_base`.

File: beluga_amcl/src/amcl_nodelet.cpp [hunk 2]
```diff
@@ -423,7 +423,10 @@ void AmclNodelet::laser_callback(ExecutionPolicy&& exec_policy, const sensor_msg
           config_.laser_max_range,
       }
               .points_in_cartesian_coordinates() |
-          ranges::views::transform([](const auto& value) { return std::make_pair(value.x(), value.y()); }) |
+          ranges::views::transform([&](const auto& p) {
+            const auto result = base_to_laser_transform * Sophus::Vector3d{p.x(), p.y(), 0};
```
Comment (glpuga) [inline beluga_amcl/src/amcl_nodelet.cpp:427]: nit: idem.

**Review by glpuga — APPROVED** [review]

File: beluga_amcl/src/amcl_node.cpp (hunk 1 above)
Comment (nahueespinosa) [reply beluga_amcl/src/amcl_node.cpp]: Definitely, done in https://github.com/Ekumen-OS/beluga/pull/276/commits/04a91463ec39ceed3ee673d828cb44c5f66a3f50.

File: beluga_amcl/src/amcl_nodelet.cpp (hunk 2 above)
Comment (nahueespinosa) [reply beluga_amcl/src/amcl_nodelet.cpp]: Done.

**Pull request merged**